### PR TITLE
[GR-40820] Ensure section offsets written to DWARF records are relocatable

### PR DESCRIPTION
**Pull request by adinn**
This patch is a fix for issue  #4856.

It ensures that any section global offsets written into DWARF section records are relocatable.

### CODE REVIEW

**Comment by adinn** [comment]
@koutheir This is the patch I mentioned which should fix the problem you were seeing.

**Comment by koutheir** [comment]
Applying this patch solves the issue #4856 for me. Thank you, @adinn!

```plain
$ gdb ./hello-aarch64
GNU gdb (GDB) 12.1
...
Reading symbols from ./hello-aarch64...
(gdb) b Hello_sub3_6f681e7c6623058e66e480d3bd5e6c6b9d5562b6 
Breakpoint 1 at 0x405734
(gdb) r
Starting program: /home/microdoc/kat/hello-aarch64 
[Thread debugging using libthread_db enabled]
Using host libthread_db library "/usr/lib/libthread_db.so.1".
[New Thread 0xfff7f7cff120 (LWP 6185)]
Hello World Java!
-> hola
-> hola2

Thread 1 "hello-aarch64" hit Breakpoint 1, 0x0000000000405734 in Hello::sub3(java.lang.String *) (__0=<optimized out>)
(gdb) bt
#0  0x0000000000405734 in Hello::sub3(java.lang.String *) (__0=<optimized out>)
#1  0x0000000000405684 in Hello::sub2(java.lang.String *) (__0=<optimized out>)
#2  0x0000000000405334 in Hello::sub1(java.lang.String *) (__0=<optimized out>)
#3  0x0000000000405058 in Hello::main(java.lang.String[] *) (__0=<optimized out>)
#4  0x0000000000413764 in com.oracle.svm.core.JavaMainWrapper::runCore0(void) () at com/oracle/svm/core/JavaMainWrapper.java:166
#5  0x00000000004133f0 in com.oracle.svm.core.JavaMainWrapper::run(int, org.graalvm.nativeimage.c.type.CCharPointerPointer *) (argc=<optimized out>, 
    argv=<optimized out>) at com/oracle/svm/core/JavaMainWrapper.java:214
#6  0x00000000004510a0 in com.oracle.svm.core.code.IsolateEnterStub::JavaMainWrapper_run_5087f5482cc9a6abc971913ece43acb471d2631b(int, org.graalvm.nativeimage.c.type.CCharPointerPointer *) (__0=<optimized out>, __1=<optimized out>) at com/oracle/svm/core/code/IsolateEnterStub.java:1
(gdb) c
Continuing.
java.lang.Throwable
        at Hello.sub3(Hello.java:18)
        at Hello.sub2(Hello.java:14)
        at Hello.sub1(Hello.java:9)
        at Hello.main(Hello.java:4)
-> hola23
[Thread 0xfffff7fbe020 (LWP 6183) exited]
[Thread 0xfff7f7cff120 (LWP 6185) exited]
[New process 6183]
[Inferior 1 (process 6183) exited normally]
```

**Comment by koutheir** [comment]
Should I close the issue?

**Comment by fniephaus** [comment]
> Should I close the associated issue? Or should I wait until this PR is merged?

We usually wait until PRs are merged.

**Review by zakkak — APPROVED** [review]
LGTM. Thanks @adinn !

**Comment by adinn** [comment]
> Applying this patch solves the issue https://github.com/oracle/graal/issues/4856 for me. Thank you, @adinn!

@koutheir Excellent! Thanks very much for retesting.

**Review by olpaw — APPROVED** [review]

**Comment by fniephaus** [comment]
Merged via https://github.com/oracle/graal/commit/f7e1ca26340a2166c80f901833a8aba52890b125.

**Pull request closed without merging**
